### PR TITLE
fix(entities):  validations for domain and type, NR-70290

### PR DESCRIPTION
**Pull request by NSSPKrishna**
# Description

Fixes # [2135](https://github.com/newrelic/terraform-provider-newrelic/issues/2135)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

## How to test this change?

Please describe how to test your changes. Include any relevant steps in the UI, HCL file(s), commands, etc

- Step 1
- Step 2
- etc

### CODE REVIEW

**Review by mbazhlekova — APPROVED** [review]

File: newrelic/data_source_newrelic_entity.go [hunk 1]
```diff
@@ -28,21 +27,19 @@ func dataSourceNewRelicEntity() *schema.Resource {
...
 			"type": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				Computed:     true,
-				Description:  "The entity's type. Valid values are APPLICATION, DASHBOARD, HOST, MONITOR, and WORKLOAD.",
-				ValidateFunc: validation.StringInSlice([]string{"APPLICATION", "DASHBOARD", "HOST", "MONITOR", "WORKLOAD"}, true),
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The entity's type. Valid values are APPLICATION, DASHBOARD, HOST, MONITOR,SERVICE and WORKLOAD.",
```
Comment (sanderblue) [inline newrelic/data_source_newrelic_entity.go:33]: Do we not have more valid types than this now? Also missing a space after the comma for `MONITOR,`.

Comment (sanderblue) [inline newrelic/data_source_newrelic_entity.go:33]: Minor, but missing a space after the comma at `MONITOR,SERVICE`

**Review by kidk — CHANGES_REQUESTED** [review]
This is missing documentation changes.

File: newrelic/data_source_newrelic_entity.go (hunk 1 above)
Comment (NSSPKrishna) [reply newrelic/data_source_newrelic_entity.go]: done

File: website/docs/d/entity.html.markdown [hunk 2]
```diff
@@ -99,3 +99,19 @@ In addition to all arguments above, the following attributes are exported:
 * `account_id` - The New Relic account ID associated with this entity.
 * `application_id` - The domain-specific application ID of the entity. Only returned for APM and Browser applications.
 * `serving_apm_application_id` - The browser-specific ID of the backing APM entity. Only returned for Browser applications.
+
+
+## Additional Example
+
+-> If the entities are not found please try again without providing the `types` field.
+
+```hcl
```
Comment (kidk) [inline website/docs/d/entity.html.markdown:108]: Can you add the ones for Otel and Lambda?

**Review by kidk — APPROVED** [review]

**Pull request merged**